code/modules/mob/living/carbon/xenomorph/mutators/strains/drone/healer.dm

```diff
@@ -173,3 +183,118 @@
+		return
+	var/image/holder = bound_xeno.hud_list[PLASMA_HUD]
+	holder.overlays.Cut()
+	var/percentage_transferred = min(round((transferred_amount / required_transferred_amount) * 100, 10), 100)
+	if(percentage_transferred)
+		holder.overlays += image('icons/mob/hud/hud.dmi', "xenoenergy[percentage_transferred]")
+
+/datum/behavior_delegate/drone_healer/handle_death(mob/M)
+	var/image/holder = bound_xeno.hud_list[PLASMA_HUD]
+	holder.overlays.Cut()
+
+/datum/action/xeno_action/activable/healer_sacrifice
+	name = "Sacrifice"
+	action_icon_state = "screech"
+	ability_name = "sacrifice"
+	var/max_range = 1
+	var/transfer_mod = 0.75 // only transfers 75% of current healer's health
+	macro_path = /datum/action/xeno_action/verb/verb_healer_sacrifice
+	action_type = XENO_ACTION_CLICK
+	ability_primacy = XENO_PRIMARY_ACTION_5
+
+/datum/action/xeno_action/verb/verb_healer_sacrifice()
+	set category = "Alien"
+	set name = "Sacrifice"
+	set hidden = TRUE
+	var/action_name = "Sacrifice"
+	handle_xeno_macro(src, action_name)
+
+/datum/action/xeno_action/activable/healer_sacrifice/use_ability(atom/atom)
+	var/mob/living/carbon/xenomorph/xeno = owner
+	var/mob/living/carbon/xenomorph/target = atom
+
+	if(!istype(target))
+		return
+
+	if(target == xeno)
+		to_chat(xeno, "You can't heal yourself!")
+		return
+
+	if(isfacehugger(target) || islesserdrone(target))
+		to_chat(xeno, "It would be a waste...")
+		return
+
+	if(!xeno.check_state())
+		return
+
+	if(target.stat == DEAD)
+		to_chat(xeno, SPAN_WARNING("[target] is already dead!"))
+		return
+
+	if(target.health >= target.maxHealth)
+		to_chat(xeno, SPAN_WARNING("[target] is already at max health!"))
+		return
+
+	if(!isturf(xeno.loc))
+		to_chat(xeno, SPAN_WARNING("You cannot transfer health from here!"))
+		return
+
+	if(get_dist(xeno, target) > max_range)
+		to_chat(xeno, SPAN_WARNING("You need to be closer to [target]."))
+		return
+
+	xeno.say(";MY LIFE FOR THE QUEEN!!!")
+
+	target.gain_health(xeno.health * transfer_mod)
+	target.updatehealth()
+
+	target.xeno_jitter(1 SECONDS)
+	target.flick_heal_overlay(3 SECONDS, "#44253d")
+
+	target.visible_message(SPAN_XENONOTICE("[xeno] explodes in a deluge of regenerative resin salve, covering [target] in it!"))
+	xeno_message(SPAN_XENOANNOUNCE("[xeno] sacrifices itself to heal [target]!"), 2, target.hive.hivenumber)
+
+	var/datum/behavior_delegate/drone_healer/behavior_delegate = xeno.behavior_delegate
+	if(istype(behavior_delegate) && behavior_delegate.transferred_amount >= behavior_delegate.required_transferred_amount && xeno.client && xeno.hive)
+		var/datum/hive_status/hive_status = xeno.hive
+		var/turf/spawning_turf = get_turf(xeno)
+		if(!hive_status.hive_location)
+			addtimer(CALLBACK(xeno.hive, TYPE_PROC_REF(/datum/hive_status, respawn_on_turf), xeno.client, spawning_turf), 0.5 SECONDS)
+		else
+			addtimer(CALLBACK(xeno.hive, TYPE_PROC_REF(/datum/hive_status, free_respawn), xeno.client), 5 SECONDS)
+
+	xeno.gib("sacrificing itself")
```

Comment:
runtime error: death called with string cause (sacrificing itself) instead of datum
```suggestion
	xeno.gib(create_cause_data("sacrificing itself", src))
```